For review

konstzv opened the pull request.

konstzv committed:
For review

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -1,12 +1,68 @@
+            text = getString(R.string.copyring)
+            gravity = Gravity.CENTER
+        }
+        main_linear_layout.addView(textView)
```
Предложение здесь еще добавить padding

konstzv replied:
Добавил

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -1,12 +1,68 @@
+
+    }
+
+    private fun addCopyring() {
```
При перевороте вью еще раз добавится или нет? Почему?

konstzv replied:
Как я понимаю добавится новая. Так как старася пропадет при повороте, ведь вызовется метод onDestroy и активити пересоздастся в onCreate

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -1,12 +1,68 @@
+    private fun openUrl(url: String) {
+        val openUrlIntent = Intent(Intent.ACTION_VIEW)
+        openUrlIntent.data = Uri.parse(url)
+        if (checkIntentResolving(openUrlIntent)) return
```
преждевременный выход из метода. холиварный вопрос, но не рекомендую делать

konstzv replied:
Ок не буду так больше
Я в ревью другом посоветовал наоборот так сделать)

xanderblinov replied:
советуй обратно)

xanderblinov replied:
все равно преждевременный выход, лучше делать if - else

konstzv replied:
Косячнул исправляя) забыл вызов одной из активити, поправил

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -1,12 +1,68 @@
+
+    }
+
+    private fun checkIntentResolving(intent: Intent): Boolean {
```
симантика метода — только проверяет, а на самом деле еще и резолвит

konstzv replied:
Исправил

xanderblinov commented on app/src/main/res/layout-land/activity_main.xml:
```diff
@@ -0,0 +1,36 @@
+        android:id="@+id/imageView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_weight="2"
```
когда ставят вес зануляют один из параметров высота / ширина

konstzv replied:
Исправил

xanderblinov commented on app/src/main/res/layout-land/activity_main.xml:
```diff
@@ -0,0 +1,36 @@
+        android:layout_height="match_parent"
+        android:layout_weight="2"
+        android:background="?android:attr/colorBackground"
+        android:contentDescription="@string/photo_content_description"
```
на практике практически никогда не указывают. поле пришло из веб

konstzv replied:
Я как понял это для тех кто плохо видит, чтоб читал голосовой помошник

xanderblinov replied:
точняк оставляем

xanderblinov commented on app/src/main/res/layout-land/activity_main.xml:
```diff
@@ -0,0 +1,36 @@
+    <ScrollView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_weight="1">
```
аналогичная проблема с весом и шириной / высотой

konstzv replied:
Исправил

xanderblinov commented on app/src/main/res/layout-land/activity_main.xml:
```diff
@@ -0,0 +1,36 @@
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical"
+            android:paddingBottom="@dimen/main_block_bottom_padding">
```
часто не используют dimen специфичный для каждого отступа, потоу что на большом проекте их будет миллион

konstzv replied:
понял
он просто в двух лейаутах одинаковый

xanderblinov commented on app/src/main/res/values-ru/strings.xml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">Загулин Константин</string>
+    <string name="send_message">ПОСЛАТЬ СООБЩЕНИЕ</string>
```
капс обычно делается не в ресурсах а в параметрах view

konstzv committed:
Added checking edit text if its blank and toast to warn user

xanderblinov commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="main_section_start_margin">50dp</dimen>
```
обычно определяются парметры типа 8dp, 16 dp и тд которыми оперируют при верстке

konstzv replied:
У нас же по тз 50dp отступ

xanderblinov replied:
это комментарий не к этому размеру, а ко всему блоку. Такие размеры да, нужно указывать точно так как ты сделал

xanderblinov commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,17 @@
 <resources>
-    <string name="app_name">MyCard</string>
+    <string name="app_name">Zagulin Konstantin</string>
+    <string name="story">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non p ...
```
МОжно было в транслэйторе перевести. Ipsum часто используется для tools

konstzv replied:
Я и в русской версии случайный текст написал. Но английский и русский не совпадают

xanderblinov reviewed (commented):
Отличная работа, добавил комментариев на подумать

konstzv committed:
Correcting caps in strings and replacing it with view attribute

konstzv committed:
Correction of layout_weight
Padding for copyring textview
changing checkIntentResolving method-> removing startActivity from it

konstzv committed:
Added startActivity after resolving ok

konstzv committed:
Added expanding toolbar

konstzv committed:
Added bottom sheet

konstzv committed:
Android x migration

konstzv committed:
Android x imports
Snackbars

konstzv committed:
atr correction

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -37,41 +72,75 @@ class MainActivity : AppCompatActivity() {
-        send_msg_btn.setOnClickListener { sendMessageFromUser() }
+
+        send_msg_btn.setOnClickListener {
+            onSendBtnClick()
```
выглядит неоднородно - надо либо везде в одну строку 6 либо везде переносить

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -1,29 +1,64 @@
+
+    private fun setBottomSheet() {
+        bottomSheetBehavior = BottomSheetBehavior.from(bottom_sheet)
+        bottomSheetBehavior!!.state = BottomSheetBehavior.STATE_HIDDEN
```
не стоит так делать. Это NPE от которого котлин ушел. надо делать через ```bottomSheetBehavior?.let{} ``` или вообще сделать bottomSheetBehavior локальной переменной

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -37,41 +72,75 @@ class MainActivity : AppCompatActivity() {
+
+    private fun onSendBtnClick() {
+        if (sendMessageFromUser()) {
+            bottomSheetBehavior!!.state = BottomSheetBehavior.STATE_HIDDEN
```
!!

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -37,41 +72,75 @@ class MainActivity : AppCompatActivity() {
+    private fun onFabCLick() {
+
+        if (bottomSheetBehavior != null) {
+            if (bottomSheetBehavior!!.state == BottomSheetBehavior.STATE_HIDDEN) {
```
!!

xanderblinov commented on app/src/main/java/com/zagulin/mycard/MainActivity.kt:
```diff
@@ -83,9 +152,25 @@ class MainActivity : AppCompatActivity() {
+
+    private fun hideInput() {
+        val imm = getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
+        if (currentFocus != null)
```
можно заменить на 
```
currentFocus?.let{
imm.hideSoftInputFromWindow(it.windowToken, 0)
}
```

konstzv replied:
Исправил

xanderblinov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,29 +1,83 @@
+
+        </com.google.android.material.appbar.AppBarLayout>
+
+        <androidx.core.widget.NestedScrollView xmlns:android="http://schemas.android.com/apk/res/android"
```
тут схема не нужна, ее добавляют обычно в корневом элементе

konstzv replied:
Не заметил, исправил

xanderblinov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,29 +1,83 @@
+        android:layout_alignParentBottom="true"
+        android:layout_alignParentEnd="true"
+        android:layout_marginBottom="20dp"
+        android:layout_marginEnd="20dp"
```
не забывай про End/Right (если проет поддерживает старые версии ОС)

konstzv replied:
Ну апи у меня выше 17-го
А для ниже типа два атрибута добавлять для каждой стороны?

xanderblinov commented on app/src/main/res/layout/bottom_sheet.xml:
```diff
@@ -0,0 +1,52 @@
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_margin="5dp"
```
странный margin. чтобы таких чтук не было советую вынести в dimen все возможные отступы стандартныеи их переиспользовать

konstzv replied:
То есть стоит в дизайне продумать несколько типов отступа типа маленький, среднйи , большой и их переиспользовать? Например этот назвать short_indent

xanderblinov commented on app/src/main/res/layout/bottom_sheet.xml:
```diff
@@ -0,0 +1,52 @@
+            android:layout_height="wrap_content"
+            android:layout_gravity="end"
+            android:layout_weight="1"
+            android:background="?android:attr/selectableItemBackground"
```
посмотри в сторону style чтобы выносить все аттрибуты и не писать их по много раз

xanderblinov reviewed (commented):
🤘

Уже круто, добавил еще комментариев, как можно что улучшить

konstzv committed:
kotlin,gradle,styles update

konstzv committed:
styles, dimension

konstzv committed:
correcting potential np, kotlin code reformat
